docs/configuration/jobs.rst

```diff
@@ -173,6 +173,74 @@ Note that you will have to trigger replication manually using the ``zrepl signal
        type: manual
      ...
 
+.. _job-snapshotting-hooks:
+
+Pre- and post-snapshot Hooks
+----------------------------
+
+Jobs with `periodic snapshots <job-snapshotting-spec_>`_ can run an executable before and/or after taking the snapshot,
+configured in an optional ``hooks`` section of the ``snapshotting`` configuration:
+
+::
+
+
+    jobs:
+    - type: push
+      filesystems: {
+        "<": true,
+        "tmp": false
+      }
+      snapshotting:
+        type: periodic
+        prefix: zrepl_
+        interval: 10m
+        hooks:
+          pre: scripts/pre-snapshot.sh
+          post: /path/to/scripts/zrepl-notify.sh
+          timeout: 30s
+          keep_inconsistent: false
+      ...
```

Comment:
`hooks` is already plural.
Why not:

```
   hooks:
    - type: script
      pre: scripts/pre-snapshot.sh
      post: /path/to/scripts/zrepl-notify.sh
      timeout: 30s
      keep_inconsistent: false
```

This would leave a syntax-compatible option for future extension with multiple hooks in the future, e.g. built-in support for locking & unlocking mysql & postgres.

We could also just have `type`, but not a list:

```
   hook:
      type: script
      pre: scripts/pre-snapshot.sh
      post: /path/to/scripts/zrepl-notify.sh
      timeout: 30s
      keep_inconsistent: false
```

Opinions on multiple hooks / future extensibility?

Reply:
I think the list would make sense if each item in the list had a `filesystem` item that specified which filesystems it operated on. I actually like that idea because it avoids the whole "here's how to filter by filesystem in your script" tutorial.

I'm going to push all the other changes and then take a crack at using the filesystem filter code in this location.